library/spdm_crypt_lib/libspdm_crypt_crypt.c

```diff
@@ -1542,6 +1542,98 @@ bool libspdm_hmac_all(uint32_t base_hash_algo, const void *data,
     return hmac_function(data, data_size, key, key_size, hmac_value);
 }
 
+/**
+ * Return HKDF extract function, based upon the negotiated HKDF algorithm.
+ *
+ * @param  base_hash_algo                 SPDM base_hash_algo
+ *
+ * @return HKDF extract function
+ **/
+libspdm_hkdf_extract_func get_spdm_hkdf_extract_func(uint32_t base_hash_algo)
+{
+    switch (base_hash_algo) {
+    case SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SHA_256:
+#if LIBSPDM_SHA256_SUPPORT == 1
+        return libspdm_hkdf_sha256_extract;
```

Comment:
Does `cryptlib.h` really need to implement, for example, `libspdm_hkdf_sha256_extract`? Why can't these functions live in `spdm_crypt_lib.h` and call, for example, `libspdm_hmac_sha256_all`?

Reply:
We should follow the crypto API. And hmac will be called by the crypto lib API, such as https://github.com/Mbed-TLS/mbedtls/blob/f71e2878084126737cc39083e1e15afc459bd93d/library/hkdf.c#L49.

I don't see the need to call hmac directly.

Reply:
Ok that'll be a separate issue on the dividing line between cryptography that libspdm manages and cryptography that `cryptlib.h` manages.